### DISCUSSION TITLE
What is ‘training Set’ and ‘test Set’ in a Machine Learning Model? How Much Data Will You Allocate for Your Training, Validation, and Test Sets?

### DISCUSSION THREAD

**Discussion by luffy-Monkey123** [luffy-Monkey123/python-cool-programs#15] [Q&A] [answered]
What is ‘training Set’ and ‘test Set’ in a Machine Learning Model? How Much Data Will You Allocate for Your Training, Validation, and Test Sets?

**Comment by RUTAGENGWA-ASANTE-BRUCE** [comment, marked as answer]
There is a three-step process followed to create a model:

1. Train the model
2. Test the model 
3. Deploy the model

<html>
<body>


Training Set | Test Set
-- | --
The training set is examples given to the model to analyze and learn70% of the total data is typically taken as the training datasetThis is labeled data used to train the model | The test set is used to test the accuracy of the hypothesis generated by the modelRemaining 30% is taken as testing datasetWe test without labeled data and then verify results with labels

<p style="box-sizing: border-box; font-weight: 400; font-style: normal; font-family: Roboto, sans-serif; margin-top: 0px; margin-bottom: 26px; color: rgb(81, 86, 94); font-size: 16px; line-height: 26px !important; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Consider a case where you have labeled data for 1,000 records. One way to train the model is to expose all 1,000 records during the training process. Then you take a small set of the same data to test the model, which would give good results in this case.</p><p style="box-sizing: border-box; font-weight: 400; font-style: normal; font-family: Roboto, sans-serif; margin-top: 0px; margin-bottom: 26px; color: rgb(81, 86, 94); font-size: 16px; line-height: 26px !important; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">But, this is not an accurate way of testing. So, we set aside a portion of that data called the ‘test set’ before starting the training process. The remaining data is called the ‘training set’ that we use for training the model. The training set passes through the model multiple times until the accuracy is high, and errors are minimized.</p><br class="Apple-interchange-newline">
</body>
</html>
Train Set
Now, we pass the test data to check if the model can accurately predict the values and determine if training is effective. If you get errors, you either need to change your model or retrain it with more data.

Test Set

Regarding the question of how to split the data into a training set and test set, there is no fixed rule, and the ratio can vary based on individual preferences.
